File: Binary Trees/Traversals/05-Inorder Traversal of Binary Tree.cpp
class Solution {
public:
    void helperFunction(TreeNode* root, vector<int>& traversal){
        if(root == nullptr){
            return;
        }

        helperFunction(root -> left, traversal);
        traversal.push_back(root -> val);
        helperFunction(root -> right, traversal);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        if(root == nullptr){
            return {};
        }

        vector<int> traversal;
        helperFunction(root, traversal);

        return traversal;
    }
};

File: Binary Trees/Traversals/05-Inorder Traversal of Binary Tree.java
class Solution {
    public void helperFunction(TreeNode root, List<Integer> traversal){
        if(root == null){
            return;
        }

        helperFunction(root.left, traversal);
        traversal.add(root.val);
        helperFunction(root.right, traversal);
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        if(root == null){
            return new ArrayList<>();
        }
        
        List<Integer> traversal = new ArrayList<>();

        helperFunction(root, traversal);

        return traversal;
    }
}